packages/cardano-services/test/load/load.test.ts

```diff
@@ -0,0 +1,288 @@
+    { name: 'Test Wallet' },
+    {
+      assetProvider: await assetProvider,
+      chainHistoryProvider: await chainHistoryProvider,
+      keyAgent: await keyAgentByIdx(0),
+      networkInfoProvider: await networkInfoProvider,
+      rewardsProvider: await rewardsProvider,
+      stakePoolProvider,
+      txSubmitProvider: await txSubmitProvider,
+      utxoProvider: await utxoProvider,
+      walletProvider: await walletProvider
+    }
+  );
+
+const runCli = (args: string[], startedString: string) =>
+  new Promise<ChildProcess>((resolve, reject) => {
+    const proc = fork(path.join(__dirname, '..', '..', 'dist', 'cli.js'), args, { stdio: 'pipe' });
+
+    const logChunk = (method: typeof logger.info, chunk: string) => {
+      for (const line of chunk.split('\n')) {
+        if (line) {
+          let msg = line;
+
+          try {
+            ({ msg } = JSON.parse(line));
+            // eslint-disable-next-line no-empty
+          } catch {}
+
+          method(`${args[0]}: ${msg}`);
+        }
+      }
+    };
+
+    proc.stderr!.once('data', (data) => reject(new Error(data.toString())));
+    proc.stderr!.on('data', (data) => logChunk(logger.error.bind(logger), data.toString()));
+    proc.stdout!.on('data', (data) => {
+      const chunk = data.toString();
+
+      logChunk(logger.info.bind(logger), chunk);
+      if (chunk.includes(startedString)) resolve(proc);
+    });
+    proc.once('error', reject);
+  });
+
+const stopProc = (proc: ChildProcess) =>
+  new Promise<void>((resolve) => (proc?.kill() ? proc.on('close', resolve) : resolve()));
+
+let serverProc: ChildProcess;
+let workerProc: ChildProcess;
+
+const startServer = async (options: TestOptions = {}) => {
+  await setupRabbitMQContainer(containerName);
+  if (env.START_LOCAL_HTTP_SERVER)
+    serverProc = await runCli(
+      [
+        'start-server',
+        '--api-url',
+        configEnv.TX_SUBMIT_HTTP_URL,
+        ...(options.directlyToOgmios ? [] : ['--use-queue']),
+        ...commonArgs,
+        ServiceNames.TxSubmit
+      ],
+      '"msg":"Started'
+    );
+};
+
+const startWorker = async (options: TestOptions = {}) => {
+  workerProc = await runCli(
+    [
+      'start-worker',
+      ...commonArgs,
+      ...(options.parallel ? ['--parallel', '--parallel-txs', env.WORKER_PARALLEL_TRANSACTION.toString()] : [])
+    ],
+    '"msg":"TxSubmitWorker: starting'
+  );
+};
+
+const stopServer = () => stopProc(serverProc);
+const stopWorker = () => stopProc(workerProc);
+
+describe('load', () => {
+  const testReports: TestReport[] = [];
+
+  let wallet: ObservableWallet;
+  let address: Cardano.Address;
+
+  const prepareWallets = async () => {
+    wallet = await getWallet();
+    ({ address } = (await firstValueFrom(wallet.addresses$))[0]);
+
+    const startTime = Date.now();
+    await firstValueFrom(wallet.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
+    logger.info(`Time to settle wallets ${Date.now() - startTime}ms`);
```

Comment:
This is moreover an information I logged to try to solve the problem I had on past Thursday, now probably it is better to remove it; anyway `console.time` logs to the console rather than using the `logger` object; that's the main reason why I didn't use it

Reply:
I think using `console` directly in tests is fine

Reply:
`process.hrtime` would yield more precision. I suspect this is being rounded down:

![image](https://user-images.githubusercontent.com/303881/173576374-f69206ca-41cb-479c-bc1f-306d41286497.png)

Reply:
The logged time at the beginning of this discussion was already removed: I logged it to solve the long lasting time to settle the wallets, but being it resolved and being that info not useful for test purposes, I removed it.

We are measuring time intervals of hundreds of milliseconds, I think that increasing time granularity to nanoseconds would be not so meaningful.

Last, the `0` time you are highlighting is not due to approximation or rounding but because rather than implementing an algorithm to check when transactions are in mempool I simply measured the time for the resolution or the `Promise` returned by the `submitTx` method; currently it is zero as that method has a _fire and forget_ behavior; once ADP-1823 will be completed (that's what I'm working on currently) this measure will become meaningful.

Reply:
Thanks for the explanation, all good from my side